core/src/test/java/org/elasticsearch/action/admin/indices/template/put/MetaDataIndexTemplateServiceTests.java

```diff
@@ -33,13 +33,43 @@
 import org.elasticsearch.test.ESTestCase;
 import org.junit.Test;
 
-import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 
 public class MetaDataIndexTemplateServiceTests extends ESTestCase {
     @Test
-    public void testIndexTemplateInvalidNumberOfShards() throws IOException {
+    public void testIndexTemplateInvalidNumberOfShards() {
+        PutRequest request = new PutRequest("test", "test_shards");
+        request.template("test_shards*");
+
+        Map<String, Object> map = Maps.newHashMap();
+        map.put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, "0");
+        request.settings(Settings.settingsBuilder().put(map).build());
+
+        List<Throwable> throwables = putTemplate(request);
+        assertEquals(throwables.size(), 1);
+        assertTrue(throwables.get(0) instanceof InvalidIndexTemplateException);
+        assertTrue(throwables.get(0).getMessage().contains("index must have 1 or more primary shards"));
+    }
+
+    @Test
+    public void testIndexTemplateValidationAccumulatesValidationErrors() {
+        PutRequest request = new PutRequest("test", "putTemplate shards");
+        request.template("_test_shards*");
+
+        Map<String, Object> map = Maps.newHashMap();
+        map.put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, "0");
+        request.settings(Settings.settingsBuilder().put(map).build());
+
+        List<Throwable> throwables = putTemplate(request);
+        assertEquals(throwables.size(), 1);
+        assertTrue(throwables.get(0) instanceof InvalidIndexTemplateException);
+        assertTrue(throwables.get(0).getMessage().contains("name must not contain a space"));
+        assertTrue(throwables.get(0).getMessage().contains("template must not start with '_'"));
+        assertTrue(throwables.get(0).getMessage().contains("index must have 1 or more primary shards"));
```

Comment:
I think assertThat(throwables.get(0).getMessage(), containsString(...)) might be a bit better, especially if we don't specify an error message ourselves. Same for assertThat(throwables.get(0), instanceOf(...)).